api/dataprocessor.go

```diff
@@ -188,6 +188,8 @@ func (s *Server) getTargets(reqs []models.Req) ([]models.Series, error) {
 	if len(localReqs) > 0 {
 		wg.Add(1)
 		go func() {
+			// the only errors returned are from us catching panics, so we should treat them
+			// all as internalServerErrors
```

Comment:
with http things I mean:
1) dataprocessor.go was moved into the api package as part of #378 , I thought `api` was supposed to be "the http api server". or should we just think of it more broadly, as the api exposed by metrictank, of which http is only one "method" ? and cli could be another?
2) this PR introduces a bunch of log lines to dataprocessor.go with the `HTTP` prefix. note that we already had the `DP` prefix in a few places.
3) the InternalServerError mentioned above: this is a http term. in other uses of the dataprocessor (such as upcoming cli tool) there may not be a server involved so it would just be an internal error.